front/src/applications/operationalStudies/views/Scenario/components/ManageTimetableItem/Itinerary/PathStepItem.tsx

```diff
@@ -16,11 +19,103 @@ type PathStepProps = {
   };
 };
 
-const PathStepItem = ({ pathStep, index, hidePathfindingLine, categoryColor }: PathStepProps) => {
+const PathStepItem = ({
+  pathStep,
+  pathStepMetadata,
+  index,
+  hidePathfindingLine,
+  categoryColor,
+}: PathStepProps) => {
   const { t } = useTranslation('operational-studies', {
     keyPrefix: 'manageTimetableItem.itineraryModal',
   });
 
+  const secondaryCodeSuggestions = useMemo(() => {
+    const secondaryCodes = [{ label: '', id: '' }];
+    if (
+      !pathStepMetadata ||
+      pathStepMetadata?.isInvalid ||
+      pathStepMetadata.type === 'trackOffset'
+    ) {
+      return [];
+    }
+    return secondaryCodes.concat(
+      Array.from(pathStepMetadata.locationsBySecondaryCode.keys()).map((key) => ({
+        label: key,
+        id: key,
+      }))
+    );
```

Comment:
nit: you can create a function `isOpRefMetadata` and simplify your code
```suggestion
    if (!isOpRefMetadata(pathStepMetadata)) return [];
    return [
      { label: '', id: '' },
      ...Array.from(pathStepMetadata.locationsBySecondaryCode.keys()).map((key) => ({
        label: key,
        id: key,
      })),
    ];
```

Reply:
Done